platform/lib/platform_web/controllers/media/export_controller.ex

```diff
@@ -74,56 +77,109 @@ defmodule PlatformWeb.ExportController do
     final_query = MediaSearch.filter_viewable(full_query, conn.assigns.current_user)
-    results = Material.query_media(final_query, for_user: conn.assigns.current_user)
-
-    max_num_versions =
-      Enum.max(
-        results
-        |> Enum.map(fn media ->
-          length(media.versions |> Enum.filter(&(&1.visibility == :visible)))
-        end),
-        fn -> 0 end
-      )
-
-    Temp.track!()
-    path = Temp.path!(suffix: "atlos-export.csv")
-    file = File.open!(path, [:write, :utf8])
 
-    fields_excluding_custom =
-      [:slug, :project, :inserted_at, :updated_at, :latitude, :longitude] ++
-        Attribute.attribute_names() ++
-        Enum.map(1..max_num_versions, &("source_" <> to_string(&1)))
-
-    # Remove "weird" fields that are redundant or not useful
-    fields_excluding_custom =
-      Enum.reject(fields_excluding_custom, fn field ->
-        field in [:geolocation]
+    results =
+      Material.query_media(final_query, for_user: conn.assigns.current_user)
+      |> List.flatten()
+      |> Enum.map(fn media ->
+        media_slug = Media.slug_to_display(media)
+        Logger.debug("Checking media #{media_slug}")
+
+        media.versions
+        |> Enum.map(fn version ->
+          Logger.debug("Checking version #{media_slug}/#{version.scoped_id}")
+          folder_name = "#{root_folder_name}/#{media_slug}/#{media_slug}-#{version.scoped_id}"
+
+          version.artifacts
+          |> Enum.map(fn artifact ->
+            location = Material.media_version_artifact_location(artifact)
+            f_extension = artifact.file_location |> String.split(".") |> List.last()
+            fname = "#{artifact.type}_#{media_slug}-#{version.scoped_id}.#{f_extension}"
+            Logger.debug("Artifact #{fname}: #{location}")
+            Zstream.entry("#{folder_name}/#{fname}", HTTPDownload.stream!(location))
+          end)
+        end)
       end)
+      |> List.flatten()
+      |> List.flatten()
+      |> Zstream.zip()
+
+    Logger.debug("Sending file: #{inspect(results)}")
+
+    conn =
+      conn
+      |> put_resp_content_type("application/zip")
+      |> put_resp_header(
+        "content-disposition",
+        "attachment; filename=\"#{root_folder_name}.zip\""
+      )
+      |> send_chunked(:ok)
 
-    formatted = Enum.map(results, &format_media(&1, fields_excluding_custom))
-    media = formatted |> Enum.map(fn {media, _} -> media end)
-
-    custom_attribute_names =
-      formatted |> Enum.map(fn {_, fields} -> fields end) |> List.flatten() |> Enum.uniq()
-
-    media
-    |> CSV.encode(
-      headers:
-        (fields_excluding_custom ++ custom_attribute_names) |> Enum.map(&format_field_name/1),
-      escape_formulas: true
-    )
-    |> Enum.each(&IO.write(file, &1))
-
-    :ok = File.close(file)
-
-    Platform.Auditor.log(:bulk_export, params, conn)
+    # Not sure what chunk size to choose
+    results
+    |> Stream.chunk_every(128)
+    |> Enum.map(fn chn ->
```

Comment:
Here is another place that you'll want to use `Stream.map` rather than `Enum.map` (which will wait for the entire thing to be processed and chunked before it starts running).

Re: the chunk size, let's play around with it empirically. If that corresponds to the number of files to send per chunk, then we might consider just using a chunk_size of 1?

Reply:
got it! Switched to `Stream.map` 
For the chunk_size, after some experimentation, I found that it mostly corresponds to the size of the zip file. The ratio is roughly 0.015MB ~ 0.017MB per chunk.